src/main/java/it/pagopa/pm/gateway/service/CcResumeStep1Service.java

```diff
@@ -122,32 +123,40 @@ private boolean isStepOneResultCodeOk(ThreeDS2Response response, PaymentRequestE
         String resultCode = response.getResultCode();
         String status = CREATED.name();
         String responseType = StringUtils.EMPTY;
-        String acsUrl = StringUtils.EMPTY;
+        String responseVposUrl = StringUtils.EMPTY;
         String correlationId = StringUtils.EMPTY;
         boolean isToAccount = false;
+        ThreeDS2ResponseElement threeDS2ResponseElement = response.getThreeDS2ResponseElement();
         switch (resultCode) {
             case RESULT_CODE_AUTHORIZED:
                 responseType = response.getResponseType().name();
                 isToAccount = true;
-                correlationId = ((ThreeDS2Authorization) response.getThreeDS2ResponseElement()).getTransactionId();
+                correlationId = ((ThreeDS2Authorization) threeDS2ResponseElement).getTransactionId();
                 break;
             case RESULT_CODE_CHALLENGE:
+                ThreeDS2Challenge challengeResponse = (ThreeDS2Challenge) threeDS2ResponseElement;
                 responseType = response.getResponseType().name();
-                acsUrl = ((ThreeDS2Challenge) response.getThreeDS2ResponseElement()).getAcsUrl();
-                correlationId = ((ThreeDS2Challenge) response.getThreeDS2ResponseElement()).getThreeDSTransId();
+                responseVposUrl = getChallengeUrl(challengeResponse);
+                correlationId = (challengeResponse.getThreeDSTransId());
```

Comment:
```suggestion
                correlationId = challengeResponse.getThreeDSTransId();
```

Reply:
Addressed in [63c1344](https://github.com/pagopa/pagopa-payment-transactions-gateway/pull/189/commits/63c1344951ba906fc72c75e3aa92cd57e915c7ed)